olive/resource_path.py

```diff
@@ -329,21 +325,93 @@ def save_to_dir(self, dir_path: Union[Path, str], name: str = None, overwrite: b
             temp_dir = Path(tempdir)
             retry_func(
                 ml_client.models.download,
                 [self.config.name],
                 {"version": self.config.version, "download_path": temp_dir},
-                max_tries=self.config.azureml_client.max_operation_retries,
-                delay=self.config.azureml_client.operation_retry_interval,
+                max_tries=max_tries,
+                delay=delay,
                 exceptions=ServiceResponseError,
             )
             new_path.parent.mkdir(parents=True, exist_ok=True)
             shutil.move(temp_dir / self.config.name / model_path.name, new_path)
         return str(new_path)
 
 
+class AzureMLModel(AzureMLResource):
+    """AzureML Model resource path."""
+
+    name = ResourceType.AzureMLModel
+
+    @staticmethod
+    def _default_config() -> Dict[str, Any]:
+        return {
+            "azureml_client": ConfigParam(
+                type_=AzureMLClientConfig, required=True, description="AzureML client config."
+            ),
+            "name": ConfigParam(type_=str, required=True, description="Name of the model."),
+            "version": ConfigParam(type_=Union[int, str], required=True, description="Version of the model."),
+        }
+
+    def get_path(self) -> str:
+        return f"azureml:{self.config.name}:{self.config.version}"
+
+    def get_aml_client_config(self) -> AzureMLClientConfig:
+        return self.config.azureml_client
+
+    def save_to_dir(self, dir_path: Union[Path, str], name: str = None, overwrite: bool = False) -> str:
+        ml_client = self.config.azureml_client.create_client()
+        return super().save_to_dir(
+            dir_path,
+            name,
+            overwrite,
+            ml_client,
+            self.config.azureml_client.max_operation_retries,
+            self.config.azureml_client.operation_retry_interval,
+        )
+
+
+class AzureMLRegistryModel(AzureMLResource):
+    """AzureML Model resource path."""
+
+    name = ResourceType.AzureMLRegistryModel
+
+    @staticmethod
+    def _default_config() -> Dict[str, Any]:
+        return {
+            "azureml_client": ConfigParam(
+                type_=AzureMLClientConfig, required=False, description="AzureML client config."
+            ),
+            "registry_name": ConfigParam(type_=str, required=True, description="Name of the registry."),
+            "name": ConfigParam(type_=str, required=True, description="Name of the model."),
+            "version": ConfigParam(type_=Union[int, str], required=True, description="Version of the model."),
+        }
+
+    def get_path(self) -> str:
+        return (
+            f"azureml://registries/{self.config.registry_name}/models/{self.config.name}/versions/{self.config.version}"
+        )
+
+    def save_to_dir(self, dir_path: Union[Path, str], name: str = None, overwrite: bool = False) -> str:
+        # azureml client
+        default_auth_params = {}
```

Comment:
nit: something like 
`azureml_client = self.config.azureml_client or AzureMLClientConfig()`
could allow us to not hardcode default values here. Can get it directly from `AzureMLClientConfig`

Reply:
This looks better for me. I doubted the defualt values for each fileds before but seems they are all good. Will update this.